### PR TITLE
Simplified Notifications

### PR DESCRIPTION
**Pull request by henry2611**

### CODE REVIEW

File: uploadservice/src/main/java/net/gotev/uploadservice/UploadTask.java [hunk 1]
```diff
@@ -522,6 +546,28 @@ private void updateNotification(UploadInfo uploadInfo, UploadNotificationStatusC
 
         if (statusConfig.message == null) return;
 
+        if (statusConfig.message.contains(Placeholders.ELAPSED_TIME) && params.notificationConfig.isSimplifiedNotification()) {
+            NotificationCompat.Builder summaryBuilder = null;
+            service.fileUploaded();
+            if (service.getFilesCount() == service.getFilesUploaded()) {
+                summaryBuilder = new NotificationCompat.Builder(service, params.notificationConfig.getNotificationChannelId())
+                        .setContentTitle("Pictures Uploaded")
```
Comment (mvignale1987) [inline uploadservice/src/main/java/net/gotev/uploadservice/UploadTask.java]: Move strings to resources, or static constants if context isnt available

File: uploadservice/src/main/java/net/gotev/uploadservice/UploadTask.java [hunk 2]
```diff
@@ -442,67 +456,81 @@ public void run() {
     /**
      * If the upload task is initialized with a notification configuration, this handles its
      * creation.
+     *
      * @param uploadInfo upload information and statistics
      */
     private void createNotification(UploadInfo uploadInfo) {
```
Comment (icapurro) [inline uploadservice/src/main/java/net/gotev/uploadservice/UploadTask.java]: What about moving these methods into a class behind an interface, so we can add different ways to handle notifications by extending the interface. This way we can avoid the if statements based on a flag to change the method behavior.

Something like:
```java
interface NotificationHandler {
  void createNotification(UploadInfo uploadInfo);
  void updateNotificationProgress(UploadInfo uploadInfo) ;
  void updateNotification(UploadInfo uploadInfo, UploadNotificationStatusConfig statusConfig);
}
```

Then this class will instantiate an implementation of `NotificationHandler` and use the methods as is doing right now, this way we can hide this responsibility from this class and provide an easy interface to generate different ways to implement notifications.

**Comment by icapurro** [comment]
It will be nice to add some screenshots of how the new simplified notification looks like as part of the PR.

File: uploadservice/src/main/java/net/gotev/uploadservice/UploadTask.java (hunk 2 above)
Comment (henry2611) [reply uploadservice/src/main/java/net/gotev/uploadservice/UploadTask.java]: I think i dont follow you here. @icapurro

**Pull request merged**